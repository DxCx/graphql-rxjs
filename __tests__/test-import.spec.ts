import { Observable } from 'rxjs';
import { makeExecutableSchema } from 'graphql-tools';
import { graphqlReactive } from '..';
import { GraphQLInt, GraphQLSchema, GraphQLObjectType } from 'graphql';
import 'jest';

const counterSource = Observable.interval(10).publishReplay(1).refCount();

describe('graphql-rxjs import tests', () => {
  it("can work with graphql-tools", () => {
    const typeDefs = `
      # Root Subscription
        type Query {
        someInt: Int
      }

      type Subscription {
        counter: Int
      }
      `;

    const resolvers = {
      Subscription: {
        counter(root, args, ctx) {
            return ctx.counterSource;
        },
      },
    };

    const scheme = makeExecutableSchema({typeDefs, resolvers});
    const query = `
      subscription {
        counter
      }
    `;

    return graphqlReactive(scheme, query, null, { counterSource })
      .bufferCount(3)
      .take(1)
      .toPromise().then((values) => {
        expect(values).toMatchSnapshot();
    });
  });

  it("also works with vanilla graphql objects", () => {
    const SubscriptionType = new GraphQLObjectType({
      name: 'Subscription',
      fields: {
        counter: {
          type: GraphQLInt,
          resolve: (root, args, ctx) => {
            return ctx.counterSource;
          },
        }
      }
    });

    const QueryType = new GraphQLObjectType({
      name: 'Query',
      fields: {
        someInt: {
          type: GraphQLInt,
          resolve: (root, args, ctx) => {
            return 123321;
          },
        }
      }
    });

    const scheme = new GraphQLSchema({
      query: QueryType,
      subscription: SubscriptionType,
    });

    const query = `
      subscription {
        counter
      }
    `;

    return graphqlReactive(scheme, query, null, { counterSource })
      .bufferCount(3)
      .take(1)
      .toPromise().then((values) => {
        expect(values).toMatchSnapshot();
    });
  });

  it("can also query static values", () => {
    const QueryType = new GraphQLObjectType({
      name: 'Query',
      fields: {
        someInt: {
          type: GraphQLInt,
          resolve: (root, args, ctx) => {
            return 123321;
          },
        }
      }
    });

    const scheme = new GraphQLSchema({
      query: QueryType,
    });

    const query = `
      query {
        someInt
      }
    `;

    return graphqlReactive(scheme, query, null, { counterSource })
      .bufferCount(3)
      .take(1)
      .toPromise().then((values) => {
        expect(values).toMatchSnapshot();
    });
  });

});
