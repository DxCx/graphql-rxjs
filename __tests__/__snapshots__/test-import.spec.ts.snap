// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql-rxjs import tests able to add reactive directives to schema 1`] = `
Array [
  Object {
    "data": Object {
      "counter": 0,
    },
  },
  Object {
    "data": Object {
      "counter": 1,
    },
  },
  Object {
    "data": Object {
      "counter": 2,
    },
  },
  Object {
    "data": Object {
      "counter": 3,
    },
  },
  Object {
    "data": Object {
      "counter": 4,
    },
  },
  Object {
    "data": Object {
      "counter": 5,
    },
  },
  Object {
    "data": Object {
      "counter": 6,
    },
  },
  Object {
    "data": Object {
      "counter": 7,
    },
  },
  Object {
    "data": Object {
      "counter": 8,
    },
  },
  Object {
    "data": Object {
      "counter": 9,
    },
  },
]
`;

exports[`graphql-rxjs import tests also works with vanilla graphql objects 1`] = `
Array [
  Object {
    "data": Object {
      "counter": 2,
    },
  },
  Object {
    "data": Object {
      "counter": 0,
    },
  },
  Object {
    "data": Object {
      "counter": 1,
    },
  },
]
`;

exports[`graphql-rxjs import tests can also query static values 1`] = `
Array [
  Object {
    "data": Object {
      "someInt": 123321,
    },
  },
]
`;

exports[`graphql-rxjs import tests can work with graphql-tools 1`] = `
Array [
  Object {
    "data": Object {
      "counter": 0,
    },
  },
  Object {
    "data": Object {
      "counter": 1,
    },
  },
  Object {
    "data": Object {
      "counter": 2,
    },
  },
]
`;
